File: C++/129. Sum Root to Leaf Numbers.cpp
class Solution {
public:
int ans=0;
vector<int> res ;
void solve(int n)
{
    if(ans == 0)
    {
        ans += n;
    }
    else {
        ans *= 10;
        ans += n;
    }
}
void inorder(TreeNode* root)
{
    if(root == NULL)
        return;    
    solve(root->val);
    if(root->left == NULL && root->right == NULL)
    {
        res.push_back(ans);
    }
    inorder(root->left);
    inorder(root->right);
    ans /=10;
}
int sumNumbers(TreeNode* root) {
    inorder(root);
    int sum =0;
    for(int i=0; i<res.size(); i++)
    {
        sum += res[i];
    }
    return sum;
}
};

File: JAVA/129. Sum Root to Leaf Numbers.java
class Solution {
    public int sumNumbers(TreeNode root) {
        return sumtree(root ,0);
    }
    public int sumtree(TreeNode root ,int n){
        if(root == null)return 0;
        n=n*10+root.val;
        if(root.right==null&&root.left==null)return n;
        return sumtree(root.right ,n)+sumtree(root.left,n);
    }
}
